Automattic/wp-calypso
# Refactor ui.queryArguments state and introduce a new reducer for the route

Creates new reducer `ui.route` which combines the two reducers `ui.route.query` and `ui.route.path` each combining the initial and current query arguments and path of the application route.

For instance if the app loaded from `/log-in?retry=1` and redirects to `/log-in`, the `ui.route` state will be:
```js
{
    query: {
        initial: { retry: "1" }
        current: {},
    },
    path: {
        initial: '/log-in',
        current: '/log-in',
    }
}
```

This change was initially introduced in https://github.com/Automattic/wp-calypso/pull/19036 so we group path and query arguments in the same reducer.

**Important**: With this patch, we also change the default caching behaviour of the `ui` state so we may persist that information for SSR.

### Testing Instructions
- Boot this branch locally

- Try accessing a few routes with arguments and for each one check that the sate is consistant:

- http://calypso.localhost:3000/log-in?redirect_to=https%3A%2F%2Fwordpress.com%2Fthemes
State should look like:
```
{
    route: {
        path: {
            initial: '/log-in',
            current: '/log-in'
        },
        query: {
            initial: {
                redirect_to: 'https://wordpress.com/themes',
                _timestamp: ...
            },
            current: {
                redirect_to: 'https://wordpress.com/themes',
                _timestamp: ...
            }
      }
}
```

- http://calypso.localhost:3000/log-in?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D50916
(or another tour you haven't done yet)
State should look like: 
```
{
    route: {
        path: {
            initial: '/log-in',
            current: '/log-in'
        },
        query: {
            initial: {
              client_id: '50916',
              redirect_to: 'https://public-api.wordpress.com/oauth2/authorize?response_type=code&client_id=50916',
              _timestamp: ...
            },
            current: {
                client_id: '50916',
                redirect_to: 'https://public-api.wordpress.com/oauth2/authorize?response_type=code&client_id=50916',
                _timestamp: ...
            }
        }
    }
}
```
- Check that tours can boot by adding `?tour=main` at the end of your url or any other tour other than `main` that you did not try yet. Some tours seems to be broken, so checking that you can complete them should not be part of this PR review.

### Reviews
